src/main/java/com/puppycrawl/tools/checkstyle/api/DetailAST.java

```diff
@@ -202,12 +202,7 @@ private void setParent(DetailAST parent) {
         do {
             instance.clearBranchTokenTypes();
             instance.parent = parent;
-            final DetailAST nextSibling = instance.getNextSibling();
-            if (nextSibling != null) {
-                nextSibling.previousSibling = instance;
```

Comment:
This code was removed because setting the parent doesn't change the previous/next of the siblings in any way. It just changes the parent only. This code has been here a while.